client/client.go

```diff
@@ -427,6 +511,10 @@ func (c *CfClient) setAnalyticsServiceClient(ctx context.Context) {
 //
 // Returns defaultValue if there is an error or if the flag doesn't exist
 func (c *CfClient) BoolVariation(key string, target *evaluation.Target, defaultValue bool) (bool, error) {
+	if !c.initializedBool {
+		c.config.Logger.Error("Error when calling BoolVariation and returning default variation: 'Client is not initialized'")
```

Comment:
I wonder should this be logged an an error or info?
So in UAT we don't initalize our client (because it shouldn't talk to itself), but we do still call BoolVariation and use the defaults.
If it were to log error the logs could get really noisy in places where we evaluation in the HTTP transaction.

Reply:
Thinking more about it - instead of logging, could we return this error?
Then the caller can choose to log it or not.
Ideally we could return a ErrNotInitalized, and make that part of the docs.  Users calling this can do something like

if errros.Is(ErrNotInitalized) {
    log.Warn(...)
} else {
    log.Info(...)
}

Reply:
Per your advice in offine discussion I will 

- info log this
- return the error (wrapping as a default variation error)